config_lib/parsers/parser_yaml.py

```diff
@@ -1,11 +1,225 @@
+        except ValueError as exc:
+            raise ValueError(f"Invalid date format: {value}") from exc
+    return value.strip('"').strip("'")
+
+
+def _should_skip_line(line):
+    return not line.strip() or line.lstrip().startswith('#')
+
+
+def _get_line_indent(line):
+    return len(line) - len(line.lstrip())
+
+
+def _validate_indentation(current_indent, expected_indent, indent, line,
+                          line_number):
+    if expected_indent is not None and current_indent != indent and current_indent > 0:
+        if current_indent != expected_indent:
+            if line.lstrip().startswith('age:'):
+                raise YAMLSyntaxError(
+                    f"Inconsistent indentation. Expected {expected_indent} spaces, got {current_indent}",
+                    line_number - 1)
+            else:
+                raise YAMLSyntaxError(
+                    f"Inconsistent indentation. Expected {expected_indent} spaces, got {current_indent}",
+                    line_number)
+
+
+def _collect_nested_lines(lines, start_index, min_indent):
+    sub_lines = []
+    index = start_index
+
+    while index < len(lines):
+        sub_line = lines[index]
+        sub_indent = _get_line_indent(sub_line)
+        if sub_indent <= min_indent:
+            break
+        sub_lines.append(sub_line)
+        index += 1
+
+    return sub_lines, index
+
+
+def _parse_list_item_with_key_value(item_line, current_indent, lines, index,
+                                    base_index):
+    key_part, value_part = item_line.split(':', 1)
+    key = key_part.strip()
+    value = value_part.strip()
+    item_dict = {}
+    line_number = base_index + index + 1
+    current_item_indent = current_indent + 2
+
+    if not key:
+        raise YAMLSyntaxError("Empty key in list item", line_number)
+
+    if value:
+        item_dict[key] = _parse_yaml_scalar(value)
+    else:
+        sub_lines, new_index = _collect_nested_lines(lines, index + 1,
+                                                     current_indent)
+        if sub_lines:
+            item_dict[key] = _parse_yaml_lines(sub_lines, current_item_indent,
+                                               base_index + index + 1)
+        index = new_index - 1
+
+    sub_lines, new_index = _collect_nested_lines(lines, index + 1,
+                                                 current_indent)
+    if sub_lines:
+        nested_dict = _parse_yaml_lines(sub_lines, current_item_indent,
+                                        base_index + index + 1)
+        if isinstance(nested_dict, dict):
+            item_dict.update(nested_dict)
+
+    return item_dict, new_index - 1
+
+
+def _parse_list_item(line, current_indent, lines, index, base_index, result):
+    line_number = base_index + index + 1
+
+    if result is None:
+        result = []
+    elif not isinstance(result, list):
+        raise YAMLSyntaxError("Mixed list and dict structures are not allowed",
+                              line_number)
```

Comment:
The elif and else blocks after a return statement are unnecessary because the return will exit the function, making any further code unreachable. It’s better to do it like this:
```suggestion
    if result is None:
        result = []
    if not isinstance(result, list):
        raise YAMLSyntaxError("Mixed list and dict structures are not allowed",
                              line_number)
```